src/views/demos/PayerAppSimulator/PayerApp.jsx

```diff
@@ -152,45 +163,84 @@ class PayerMobile extends React.Component {
+                ];
                 return (
-                    <Row>
-                        <Col span={24} className='text-center'>
-                            {
-                                this.state.transfersResponse && this.state.transfersResponse.transferState === 'COMMITTED'
-                                    ? (
-                                        <Result
-                                            status='success'
-                                        />
-                                    )
-                                    : (
-                                        <Result
-                                            status='error'
-                                        />
-                                    )
-                            }
-                        </Col>
-                    </Row>
+                    this.state.transfersResponse && this.state.transfersResponse.transferState === 'COMMITTED'
+                        ? (
+                            <Result
+                                extra={restart}
+                                status='success'
+                            />
+                        )
+                        : (
+                            <Result
+                                extra={restart}
+                                status='error'
+                            />
+                        )
                 );
             default:
                 return null;
         }
     };
 
-    _constructStateFromResponse = responseData => {
+    _constructStateFromResponse = (responseData, { traceId, traceUrl } = {}, type = 'error') => {
+        switch (type) {
+            case 'start':
+                return {
+                    partyInfo: { ...responseData?.getPartiesResponse?.body?.party, traceId, traceUrl },
+                    currentState: responseData?.currentState,
+                    transferId: responseData?.transferId,
+                };
+            case 'acceptParty':
+                return {
+                    fxQuoteResponse: { ...responseData?.fxQuoteResponse?.body, traceId, traceUrl },
+                    currentState: responseData?.currentState,
+                    transferId: responseData?.transferId,
+                };
+            case 'acceptConversion':
+                return {
+                    quoteResponse: { ...responseData?.quoteResponse?.body, traceId, traceUrl },
+                    currentState: responseData?.currentState,
+                    transferId: responseData?.transferId,
+                };
+            case 'acceptQuote':
+                return {
+                    transfersResponse: { ...responseData?.fulfil?.body, traceId, traceUrl },
+                    currentState: responseData?.currentState,
+                    transferId: responseData?.transferId,
+                };
```

Comment:
Hi @kalinkrustev , I think sometimes if we set autoacceptParty or autoacceptQuote, we won't get the intermediate states. For example if autoacceptParty is set to true in SDK, then we don't get partyInfo in start step, we will get both partyInfo and fxQuoteResponse in the next step. And for RECIEVE amount type, we get quoteResponse first and then fxQuoteResponse. And also there is one more type `acceptQuoteOrConversion`. I think its better to be not too specific about constructing the state from the response type. Whaterver information available in the reponse, just pull it in.

Reply:
That's why this step was handled in a generic way irrespective of the state of the transfer.